src/app.py

```
@@ -589,6 +596,69 @@ def serve_index(path: str) -> FileResponse:
     return FileResponse(Path(__file__).parent / "mqueryfront/dist/index.html")
 
 
+@app.post(
+    "/api/queue/{ursadb_id}",
+    response_model=StatusSchema,
+    tags=["queue"],
+    dependencies=[Depends(can_manage_queues)],
+)
+def add_files_to_queue(
+    ursadb_id: str, file_paths: List[FileToQueueSchema] = Body(...)
+) -> StatusSchema:
+    db.add_queued_file(ursadb_id, file_paths)
+
+    return StatusSchema(status="ok")
+
+
+@app.get(
+    "/api/queue/{ursadb_id}",
+    response_model=QueueStatusSchema,
+    tags=["queue"],
+    dependencies=[Depends(can_view_queues)],
+)
+def get_queue_status(ursadb_id: str):
+    queue_files = db.get_queued_files(ursadb_id)
+
+    if queue_files:
+        oldest_file = min(queue_files, key=lambda x: x.created_at)
+        newest_file = max(queue_files, key=lambda x: x.created_at)
+
+        return QueueStatusSchema(
+            ursadb_id=ursadb_id,
+            size=len(queue_files),
+            oldest_file=EdgeOfFileSchema(
+                path=oldest_file.path,
+                created_at=oldest_file.created_at,
+            ),
+            newest_file=EdgeOfFileSchema(
+                path=oldest_file.path,
+                created_at=newest_file.created_at,
+            ),
+        )
+    raise HTTPException(
+        status_code=400,
+        detail="Ursadb_id not found.",
+    )
```

I don't think this should raise an exception if there are no files in the queue. Unfortunately, this means that oldest_file and newest_file must be nullable (they won't have a value if a queue is empty)